src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectSnapshot.cs

```diff
@@ -58,25 +48,36 @@ public ProjectSnapshot(ProjectState state)
 
     public ProjectWorkspaceState ProjectWorkspaceState => State.ProjectWorkspaceState;
 
-    public virtual IDocumentSnapshot? GetDocument(string filePath)
+    public bool ContainsDocument(string filePath)
     {
-        lock (_lock)
+        lock (_gate)
         {
-            if (!_documents.TryGetValue(filePath, out var result) &&
-                State.Documents.TryGetValue(filePath, out var state))
-            {
-                result = new DocumentSnapshot(this, state);
-                _documents.Add(filePath, result);
-            }
-
-            return result;
+            return _filePathToDocumentMap.ContainsKey(filePath) ||
+                   State.Documents.ContainsKey(filePath);
```

Comment:
Nit: Seems like just checking `State.Documents` should be enough.

Reply:
Huh. Yeah, I guess that's true. :smile:

Reply:
Oh, good catch! Comment would be very useful here.